antsibull/docs_parsing/routing.py

```diff
@@ -59,35 +69,39 @@ def calculate_fqcns(basename, file_path, plugin_type, rel_path, path, collection
+def find_symlink_redirects(collection_name: str,
+                           plugin_type: str,
+                           directory_path: str
+                           ) -> t.DefaultDict[str, t.Dict]:
     """
-    Scans plugin folders for symbolic links, and updates plugin routing information accordingly.
+    Finds plugin redirects that are defined by symlinks.
+
+    :collection_name: FQCN of the collection we're searching within
+    :plugin_type: Type of plugins that we're scanning
+    :directory_path: Full path to the directory we're scanning.
+    :returns: Defaultdict mapping fqcn of the alias names to fqcn of the canonical plugin names.
     """
-    for plugin_type in DOCUMENTABLE_PLUGINS:
-        directory_name = 'modules' if plugin_type == 'module' else plugin_type
-        directory_path = os.path.join(collection_metadata.path, 'plugins', directory_name)
-        plugin_type_routing = plugin_routing[plugin_type]
-        if os.path.isdir(directory_path):
-            for path, _, files in os.walk(directory_path):
-                rel_path = os.path.relpath(path, directory_path)
-                for filename in files:
-                    basename, ext = os.path.splitext(filename)
-                    if ext != '.py' and not (plugin_type == 'module' and ext == '.ps1'):
-                        continue
-                    file_path = os.path.join(path, filename)
-                    if not os.path.islink(file_path):
-                        continue
-                    src_fqcn, dst_fqcn = calculate_fqcns(basename, file_path, plugin_type, rel_path,
-                                                         path, collection_name)
-                    if src_fqcn not in plugin_type_routing:
-                        plugin_type_routing[src_fqcn] = {}
-                    if 'redirect' not in plugin_type_routing[src_fqcn]:
-                        plugin_type_routing[src_fqcn]['redirect'] = dst_fqcn
-                    if plugin_type_routing[src_fqcn]['redirect'] == dst_fqcn:
-                        plugin_type_routing[src_fqcn]['redirect_is_symlink'] = True
+    plugin_type_routing = defaultdict(dict)
+    if os.path.isdir(directory_path):
+        for path, _, files in os.walk(directory_path):
+            rel_path = os.path.relpath(path, directory_path)
+            for filename in files:
+                src_basename, ext = os.path.splitext(filename)
+                if ext != '.py' and not (plugin_type == 'module' and ext == '.ps1'):
+                    continue
+
+                file_path = os.path.join(path, filename)
+                if not os.path.islink(file_path):
+                    continue
+
+                dst_basename = os.path.splitext(os.readlink(file_path))[0]
+
+                src_fqcn, dst_fqcn = calculate_plugin_fqcns(collection_name, src_basename,
+                                                            dst_basename, rel_path)
+
+                plugin_type_routing[src_fqcn]['redirect'] = dst_fqcn
```

Comment:
This needs to be changed to
```.py
                 plugin_type_routing[src_fqcn] = dst_fqcn
```
and the type of `plugin_type_routing` to `Dict[str, str]` to fix the bug mentioned above.

With both these changes, it works as expected. I'll push a commit to fix it.

Reply:
Fixed in bac0aca263ba6aeb8074623e5cff17356b41befb.